Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -213,12 +217,237 @@ private IEnvironment GetEnvironment(IHttpContextAccessor accessor, IEnvironment
+                        {
+                            //
+                            // For legacy war deployments, the only path allowed is webapps/<directory-name>
+                            //
+
+                            if (!OneDeployHelper.EnsureValidPath(artifactType, OneDeployHelper.WwwrootDirectoryRelativePath, ref path, out error))
+                            {
+                                return StatusCode400(error);
+                            }
+
+                            if (!OneDeployHelper.IsLegacyWarPathValid(path))
+                            {
+                                return StatusCode400($"path='{path}' is invalid. When type={artifactType}, the only allowed paths are webapps/<directory-name> or /home/site/wwwroot/webapps/<directory-name>. " +
+                                                     $"Example: path=webapps/ROOT or path=/home/site/wwwroot/webapps/ROOT");
+                            }
+
+                            deploymentInfo.TargetRootPath = Path.Combine(_environment.WebRootPath, path);
+                            deploymentInfo.Fetch = LocalZipHandler;
+
+                            // Legacy war deployment is equivalent to wardeploy
+                            // So always do clean deploy.
+                            deploymentInfo.CleanupTargetDirectory = true;
+                            artifactType = ArtifactType.Zip;
+                        }
+                        else
+                        {
+                            // For type=war, if no path is specified, the target file is app.war
+                            deploymentInfo.TargetFileName = "app.war";
+                        }
+
+                        break;
+
+                    case ArtifactType.Jar:
+                        if (!OneDeployHelper.EnsureValidStack(artifactType, new List<string> { OneDeployHelper.JavaSE }, ignoreStack, out error))
+                        {
+                            return StatusCode400(error);
+                        }
+
+                        deploymentInfo.TargetFileName = "app.jar";
+                        break;
+
+                    case ArtifactType.Ear:
+                        if (!OneDeployHelper.EnsureValidStack(artifactType, new List<string> { OneDeployHelper.JBossEap }, ignoreStack, out error))
+                        {
+                            return StatusCode400(error);
+                        }
+
+                        deploymentInfo.TargetFileName = "app.ear";
+                        break;
+
+                    case ArtifactType.Lib:
+                        if (!OneDeployHelper.EnsureValidPath(artifactType, OneDeployHelper.LibsDirectoryRelativePath, ref path, out error))
+                        {
+                            return StatusCode400(error);
+                        }
+
+                        deploymentInfo.TargetRootPath = OneDeployHelper.GetAbsolutePath(_environment, OneDeployHelper.LibsDirectoryRelativePath);
```

Comment:
Can you add a comment this needs to be implemented, since we no longer copy files